cmd/test-server/main.go

```diff
@@ -58,26 +61,56 @@ type tcpProbeResult struct {
 }
 
 type tcpResult struct {
-	Destination string
-	TimesInms   []float64
+	Port      int
+	TimesInms []float64
 	MinRtt    float64
 	AvgRtt    float64
 	MaxRtt    float64
 	StdDevRtt float64
 }
 
+type tcpStruct struct {
+	IP     string
+	Probes []tcpResult
+}
+
 type Results struct {
-	UUID     string
-	IPaddr   string
-	IcmpPing []RtItem
-	TcpPing  []tcpResult
+	UUID        string
+	IPaddr      string
+	Timestamp   string
+	IcmpPing    []RtItem
+	AvgIcmpStat float64
+	TcpPing     []tcpStruct
+	AvgTcpStat  float64
 }
 
 // Implementing this since Golang time.Milliseconds() function only returns an int64 value
 func fmtTimeMs(value time.Duration) float64 {
 	return (float64(value) / float64(time.Millisecond))
 }
 
+func increment(ip net.IP) {
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]++
+		if ip[j] > 0 {
+			break
+		}
+	}
+}
+
+func getAdjacentIPs(clientIP string) []string {
+	var requiredSubnet = clientIP + "/24"
+	var adjIPs []string
+	ip, ipnet, err := net.ParseCIDR(requiredSubnet)
+	if err != nil {
+		log.Fatal("Error getting adjacent IPs", err)
```

Comment:
In more complex software, you would want to return an `error` instead of terminating execution, and let the caller(s) figure out how to deal with the problem. For now, this is fine though.